extensions/control-plane/callback/callback-http-dispatcher/src/main/java/org/eclipse/edc/connector/callback/dispatcher/http/GenericHttpDispatcherDelegate.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.callback.dispatcher.http;
+
+import okhttp3.Request;
+import okhttp3.Response;
+import org.eclipse.edc.spi.types.domain.message.RemoteMessage;
+
+import java.util.function.Function;
+
+/**
+ * Generic Http dispatcher delegate that handle a single type of message
+ */
```

Comment:
Doc could be more descriptive: A delegate that dispatches HTTP requests for a message type.